### PR TITLE
Fix setState unused arguments

### PR DESCRIPTION
**Pull request by vincentmr**
**Context:**
SetState and SetBasisState cause [failures](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/10404526771/job/28813218681?pr=834#step:7:102) in L-Kokkos tests because they have unused arguments.

**Description of the Change:**
Remove argument names.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by erick-xanadu — APPROVED** [review]

File: runtime/include/QuantumDevice.hpp [hunk 1]
```diff
@@ -165,7 +165,7 @@ struct QuantumDevice {
      * @param state A state vector of size 2**len(wires)
      * @param wires The wire(s) the operation acts on
      */
-    virtual void SetState(DataView<std::complex<double>, 1> &state, std::vector<QubitIdType> &wires)
+    virtual void SetState(DataView<std::complex<double>, 1> &, std::vector<QubitIdType> &)
```
Comment (dime10) [inline runtime/include/QuantumDevice.hpp:168]: Didn't know this was possible in a function definition 😅 

(Also please clean up the PR description, the first part is not meant to be posted 😅 )

Comment (vincentmr) [reply runtime/include/QuantumDevice.hpp:168]: Do you mean "Context"?

Comment (vincentmr) [reply runtime/include/QuantumDevice.hpp:168]: > Didn't know this was possible in a function definition 😅

Just learned that from @erick-xanadu this week 😅

Comment (dime10) [reply runtime/include/QuantumDevice.hpp:168]: > Do you mean "Context"?

The part with the checkboxes 😄 Erick edited it already :)

**Pull request merged**